serverless/src/adminGetPreferencesMetrics/handler.js

```diff
@@ -0,0 +1,163 @@
+import { getDbConnection } from '../util/database/getDbConnection'
+import { getApplicationConfig } from '../../../sharedUtils/config'
+import { parseError } from '../../../sharedUtils/parseError'
+
+/**
+ * Sort the values for each key and pull out the top 5 values
+ * @param {Object} entries Object containing all preference values for a key
+ * @returns Returns the top 5 values with their counts
+ */
+const retrieveTop5Values = (entries) => {
+  const sortedEntries = Object.entries(entries).sort((a, b) => b[1] - a[1])
+
+  if (sortedEntries.length <= 5) {
+    return sortedEntries
+  }
+
+  return sortedEntries.slice(0, 6)
+}
+
+/**
+ * Retrieve all the preferences metrics for the authenticated user
+ * @param {Object} event Details about the HTTP request that it received
+ * @param {Object} context Methods and properties that provide information about the invocation, function, and execution environment
+ */
+const adminGetPreferencesMetrics = async (event, context) => {
```

Comment:
This feels very inefficient. You are pulling data for every user out of the database, then looping and sorting the entire list multiple times. For a user list as big as prod that isn't going to work well.

I think it is worth looking at using better SQL to offload all of this logic onto the database.

Reply:
Tested on 300,000 users. Loaded in 2 sec.